stablehlo/tests/compat_downgrade.mlir

```diff
@@ -0,0 +1,10 @@
+// RUN: stablehlo-opt --stablehlo-legalize-to-versionedhlo --versionedhlo-to-version='target=minimum' %s | FileCheck %s
```

Comment:
Let's have separate tests for separate passes to improve coverage.

For testing the legalization passes, we can outright copy/paste https://github.com/tensorflow/mlir-hlo/blob/master/tests/Dialect/mhlo/stablehlo-legalize-to-hlo.mlir into `stablehlo-legalize-to-vhlo.mlir` and adjust as needed. (Same for the other direction of the legalization).

Testing to-version is trickier:
  * I'd propose having one test file per version, i.e. `vhlo-to-version-v0.3.0.mlir`, `vhlo-to-version-v0.3.1.mlir`, etc, and each of those files would test `vhlo-to-version=v0.3.0`, `vhlo-to-version=v0.3.1`, etc (these tests would exercise only the ops which changed in this particular version, not the entire opset). 
  * Then we could have `vhlo-to-version.mlir` that tests generic machinery like the `invalid_program_unknown_op` below.
  * I don't think we should test `minimum` or `current`, since they are just aliases to concrete versions, and I don't think that we should burden humans with remembering which versions these aliases are referring to.

Reply:
A few thoughts:
- Opset changes should be accompanied by minor bumps. Given that, maybe we name them `vhlo-to-version-v0.3.mlir`, etc.
- What do we do for opset bumps when new ops are introduced? `vhlo-to-version-0.4.mlir` would have a single negative test?
- I like the idea of having separate test files for the sake of exhaustively testing each upgrade/downgrade, but that seems like it would accumulate over time, let's discuss _what_ versions are. Should we do a single change to a single version, or if we have 5 PR's ready with modifications, do some sort of batching?
- I agree with most of this, aside from _"I don't think we should test `current`"_. This is actually the entirety of how LLVM tests backward compatibility:
  + ([LLVM Backward Compat Policy](https://llvm.org/docs/DeveloperPolicy.html#ir-backwards-compatibility)
  + [LLVM Compatibility Test Files `compatibility.*`](https://github.com/llvm/llvm-project/tree/304f1d59ca41872c094def3aee0a8689df6aa398/llvm/test/Bitcode)
  + [Example of changes to a compat test over time](https://github.com/llvm/llvm-project/commits/304f1d59ca41872c094def3aee0a8689df6aa398/llvm/test/Bitcode/compatibility-3.8.ll). Note that filecheck's are allowed to change over time but the IR is not.


All-in-all, I propose that we separate this into another ticket that we address in the near future, I think this needs a bit more discussion and doesn't need to preclude submission of this CL. The tests in this CL are sufficient for where we stand right now, but I agree they don't scale well.

Reply:
"For testing the legalization passes"

I added a legalization pass, then it occurred to me that we probably don't need separate legalization tests. Given the 1:1 guaranteed parity, we would have identical test files that effectively accomplish what a round trip test would. So I made a `legalize_stablehlo_to_vhlo.mlir` file, and tested using FileCheck to get to VHLO, then tested round-trip StableHLO->VHLO->StableHLO to ensure that VHLO->StableHLO is working as expected.